x/erc20/keeper/ibc_callbacks.go

```diff
@@ -161,6 +165,11 @@ func (k Keeper) OnTimeoutPacket(ctx sdk.Context, _ channeltypes.Packet, data tra
 
 // ConvertCoinToERC20FromPacket converts the IBC coin to ERC20 after refunding the sender
 func (k Keeper) ConvertCoinToERC20FromPacket(ctx sdk.Context, data transfertypes.FungibleTokenPacketData) error {
+	// Record the failed ibc attempt
+	defer func() {
+		telemetry.IncrCounter(1, types.ModuleName, "ibc", "error", "total")
+	}()
+
```

Comment:
Hm why is this moved here? 🤔 with this being placed here we are also incrementing the counter for IBC errors e.g. in the case we are returning `nil` and no error - is that what we want?

Reply:
This function is only executed when IBC timeout or an Error ACK happens. 
I moved it up so that no matter the reason, it will log the failed attempt.

Reply:
ahh okay I see, that makes sense - could you add this as a note in the comment / in the doc string of the method? It wasn't apparent from the review that it's always the error case when this method is called

Reply:
After discussion with @Vvaradinov, i moved the telemetry to the only err case in the function.
This way we will only record if the conversion failed.

The previous version will record EVERY fail IBC tx, causing unnecesary overhead. Since we didnt know if the root cause of the fail tx was an erc20 conversion